### PR TITLE
Named router views

### PR DESCRIPTION
**Pull request by pleek91**
# Description
Adds the ability for a route to define multiple components rather than a single component. These components can then be rendered using named router views. A router view with no name will render the `default` component which is either `route.component` or `route.components.default`.

Closes https://github.com/kitbagjs/router/issues/174

Needs docs

**main.ts**
```ts
const routes = createRoutes([
  {
    name: 'parent',
    path: '/',
    components: {
      default: { template: '<p>default</p>' },
      one: { template: '<p>one</p>' },
      two: { template: '<p>two</p>' },
    },
  },
])

...
```
**App.vue**
```vue
<template>
    <RouterView name="one" />
    <RouterView />
    <RouterView name="two" />
</template>
```
**Rendered html** 
```html
<p>one</p>
<p>default</p>
<p>two</p>
```

### CODE REVIEW

File: src/components/routerView.browser.spec.ts [hunk 1]
```diff
@@ -188,6 +189,10 @@ test('Renders custom genericRejection component when the initialUrl does not mat
     },
   })
 
+  if (!isRouteWithComponent(router.route.matched)) {
+    throw 'Matched route does not have a single component'
+  }
```
Comment (pleek91) [inline src/components/routerView.browser.spec.ts:194]: I'm not 100% sold on the types here. See `routeProps.ts` for why this is necessary

File: src/components/routerView.vue
Comment (pleek91) [inline src/components/routerView.vue:1]: This PR is branched off of https://github.com/kitbagjs/router/pull/184 so there are a few unrelated changes that will go away once that is merged.

File: src/types/routeProps.ts [hunk 2]
```diff
@@ -30,10 +30,24 @@ type WithHooks = {
...
+   * Multiple components for named views, which can be either synchronous or asynchronous components.
+   */
+  components: Record<string, Component>,
+}
+
 /**
  * Represents properties common to parent routes in a route configuration, including hooks, path, and optional query parameters.
  */
-export type ParentRouteProps = WithHooks & {
+export type ParentRouteProps = Partial<WithComponent | WithComponents> & WithHooks & {
```
Comment (pleek91) [inline src/types/routeProps.ts:50]: Wrote this in a way where you can define a `component` property or a `components` property but not both. Which means if you want to do anything with either of those properties you have to type guard it first to determine which one is available. 

I think this is a good devx when defining routes using `createRoutes`. But its a little annoying if you need to access the component from `route.match.component` or `route.match.components` because you have to type guard it. But I don't expect users to be accessing those properties. So I think its safe to take on that slight annoyance in our internal codebase to give users a good devx when defining routes.

**Review by stackoverfloweth — APPROVED** [review]
